pyfr/__main__.py

```diff
@@ -108,6 +87,14 @@ def main():
         p.add_argument('--progress', '-p', action='store_true',
                        help='show a progress bar')
 
+    # Plugin commands
+    for scls in subclasses(BasePlugin):
```

Comment:
I would have a separate class hierarchy for this such as `BaseCLIPlugin` as this avoids awkwardness with `__init__`.